skbio/tree/_nj.py

```diff
@@ -175,10 +175,11 @@ def _compute_q(dm):
     """
     q = np.zeros(dm.shape)
     n = dm.shape[0]
-    for i in range(n):
-        for j in range(i):
-            q[i, j] = q[j, i] = \
-                ((n - 2) * dm[i, j]) - dm[i].sum() - dm[j].sum()
+    dmv = dm.to_data_frame().values
```

Comment:
Does pandas not special case the `DataFrame(np.array)` load? There are some calls here which leverage some `DataFrame` methods although they can likely be rewritten using numpy. I didn't flag this on the first PR as I had (perhaps naively) as pandas appears to simply operate on the array directly w/o copy:

```python
In [33]: a = np.arange(25).reshape(5,5)

In [34]: b = pd.DataFrame(a)

In [35]: a[0, :] = 10

In [36]: b
Out[36]:
    0   1   2   3   4
0  10  10  10  10  10
1   5   6   7   8   9
2  10  11  12  13  14
3  15  16  17  18  19
4  20  21  22  23  24

In [37]: a
Out[37]:
array([[10, 10, 10, 10, 10],
       [ 5,  6,  7,  8,  9],
       [10, 11, 12, 13, 14],
       [15, 16, 17, 18, 19],
       [20, 21, 22, 23, 24]])
```